src/main/java/book/store/bookstore/controller/BookController.java

```diff
@@ -29,7 +35,24 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto createBookRequestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto createBookRequestDto) {
         return bookService.save(createBookRequestDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBook(@PathVariable Long id,
+                              @RequestBody @Valid CreateBookRequestDto createBookRequestDto) {
+        return bookService.update(id, createBookRequestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteBook(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @GetMapping("/search")
+    public List<BookDto> search(BookSearchParameters bookSearchParameters) {
+        return bookService.search(bookSearchParameters);
+    }
```

Comment:
How are these methods connected with the task?
![image](https://github.com/k1llzers/book-store/assets/20326858/e13f01af-3ea8-4aa4-a622-9070562fc8dc)

Reply:
Yeah, I`m sorry, it was necessary to deal with github, now fixed it)